docs/src/test-advanced.md

```diff
@@ -200,6 +201,57 @@ export const test = base.extend<{ saveLogs: void }>({
 });
 ```
 
+## Launching a development web server during the tests
+
+To launch a web server during the tests, use the `webServer` option in the [configuration file](#configuration-object).
+
+This will automatically launch and wait for the web server to be ready before the tests will start:
+
+```js js-flavor=ts
+// playwright.config.ts
+import { PlaywrightTestConfig } from '@playwright/test';
+
+const config: PlaywrightTestConfig = {
+  webServer: {
+    command: 'npm run start',
+    timeout: 120 * 1000,
+  },
+  projects: [
+    {
+      name: 'Chrome',
+      use: {
+        browserName: 'chromium',
+      },
+    },
+  ],
+};
+
+export default config;
+```
+
+```js js-flavor=js
+// @ts-check
+/** @type {import('@playwright/test').PlaywrightTestConfig} */
+const config = {
+  webServer: {
+    command: 'npm run start',
+    timeout: 120 * 1000,
+  },
+  projects: [
+    {
+      name: 'Chrome',
+      use: {
+        browserName: 'chromium',
+      },
+    },
+  ],
+};
+
+mode.exports = config;
+```
+
+If your dev web-server does not print the listening URL to the console, you can also manually specify a `port` or additional environment variables, see [here](#configuration-object).
```

Comment:
Let's be more specific about "listening URL to the console" - give an example that we'll detect.